tests/components/pi_hole/test_init.py

```diff
@@ -97,3 +95,68 @@ def mock_pihole_data_call(Hole):
         hass.states.get("sensor.custom_ads_blocked_today").name
         == "Custom Ads Blocked Today"
     )
+
+
+async def test_disable_service_call(hass):
+    """Test disable service call with no Pi-hole named."""
+    with patch("homeassistant.components.pi_hole.Hole") as _hole:
+        mock_disable = Mock(return_value=Future())
```

Comment:
We can use `asynctest.CoroutineMock`:
https://asynctest.readthedocs.io/en/latest/asynctest.mock.html#asynctest.CoroutineMock